bundles/com.espressif.idf.launch.serial.core/src/com/espressif/idf/launch/serial/internal/SerialFlashLaunchConfigDelegate.java

```diff
@@ -150,11 +151,13 @@ protected void launchInternal(ILaunchConfiguration configuration, String mode, I
 		{
 			return;
 		}
-		String arguments = configuration.getAttribute(IDFLaunchConstants.ATTR_SERIAL_FLASH_ARGUMENTS, espFlashCommand);
-		arguments = varManager.performStringSubstitution(arguments);
-		if (!arguments.isEmpty())
+		String args = configuration.getAttribute(IDFLaunchConstants.ATTR_SERIAL_FLASH_ARGUMENTS, espFlashCommand);
+
+		if (!args.isEmpty())
```

Comment:
Possible NPE!

Reply:
Hi @alirana01, thanks for the review. I don't see NPE here. If the attribute's value is null, we will provide espFlashCommand, which is a List